src/vorta/views/partials/password_input.py

```diff
@@ -0,0 +1,147 @@
+from PyQt5.QtWidgets import QAction, QFormLayout, QLabel, QLineEdit, QWidget
+from vorta.i18n import translate
+from vorta.views.utils import get_colored_icon
+
+
+class PasswordLineEdit(QLineEdit):
+    def __init__(self, *, parent=None, show_visibility_button=True):
+        super().__init__(parent)
+
+        self._show_visibility_button = show_visibility_button
+        self._error_state = False
+        self._visible = False
+
+        self.setEchoMode(QLineEdit.Password)
+
+        if self._show_visibility_button:
+            self.showHideAction = QAction(self.tr("Show password"), self)
+            self.showHideAction.setCheckable(True)
+            self.showHideAction.toggled.connect(self.toggle_visibility)
+            self.showHideAction.setIcon(get_colored_icon("eye"))
+            self.addAction(self.showHideAction, QLineEdit.TrailingPosition)
+
+    def get_password(self):
+        """Return password text"""
+        return self.text()
+
+    @property
+    def visible(self):
+        """Return password visibility"""
+        return self._visible
+
+    @visible.setter
+    def visible(self, value):
+        """Set password visibility"""
+        if not isinstance(value, bool):
+            raise TypeError("visible must be a boolean value")
+        self._visible = value
+        self.setEchoMode(QLineEdit.Normal if self._visible else QLineEdit.Password)
+
+        if self._show_visibility_button:
+            if self._visible:
+                self.showHideAction.setIcon(get_colored_icon("eye"))
+            else:
+                self.showHideAction.setIcon(get_colored_icon("eye-slash"))
+
+    def toggle_visibility(self):
+        """Toggle password visibility"""
+        self.visible = not self._visible
+
+    @property
+    def error_state(self):
+        """Return error state"""
+        return self._error_state
+
+    @error_state.setter
+    def error_state(self, error):
+        """Set error state and update style"""
+        self._error_state = error
+        if error:
+            self.setStyleSheet('border: 2px solid red;')
+        else:
+            self.setStyleSheet('')
+
+
+class PasswordInput(QObject):
+    def __init__(self, *, parent=None, minimum_length=9, show_error=True, label: list = None):
+        super().__init__(parent)
+        self._minimum_length = minimum_length
+        self._show_error = show_error
+
+        if label:
+            self._label_password = QLabel(label[0])
+            self._label_confirm = QLabel(label[1])
+        else:
+            self._label_password = QLabel(self.tr("Enter passphrase:"))
+            self._label_confirm = QLabel(self.tr("Confirm passphrase:"))
+
+        # Create password line edits
+        self.passwordLineEdit = PasswordLineEdit()
+        self.confirmLineEdit = PasswordLineEdit()
+        self.validation_label = QLabel("")
+
+        self.passwordLineEdit.editingFinished.connect(self.validate)
+        self.confirmLineEdit.textChanged.connect(self.validate)
```

Comment:
Why did you decide to connect to `editingFinished` for the first entry?

Reply:
So it won't show error and red borders while user is still typing first password. Would textChanged make more sense?

Reply:
> it won't show error and red borders while user is still typing first password.

That makes sense. However after typing a password that is too short the red border won't go away (immediately) when extending the password to over 8 letters.

Reply:
@real-yfprojects What do you propose? We can either connect `textChanged` instead or perhaps replace it after first edit?

```
self.passwordLineEdit.editingFinished.connect(self.on_editing_finished)

def on_editing_finished(self):
    self.passwordLineEdit.editingFinished.disconnect()
    self.passwordLineEdit.textChanged.connect(self.validate)
    self.validate()
```